artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/QueueImpl.java

```diff
@@ -3159,6 +3162,9 @@ public void run() {
                         connection.killMessage(server.getNodeID());
                         remotingService.removeConnection(connection.getID());
                         connection.fail(ActiveMQMessageBundle.BUNDLE.connectionsClosedByManagement(connection.getRemoteAddress()));
+                        //break once a consumer gets killed. This can prevent all
+                        //consumers to this queue get killed all at once.
+                        break;
```

Comment:
This fix was using the definition of the JIRA and literally fixing the test, not the overal issue.

The proper fix would be to do some calculation with threshold using consumerSet.size();

I have a small change on QueueIMpl that I think would fix it properly, including both Kill and Notify.

```java
         Set<Consumer> consumersSet = getConsumers();

         if (consumersSet.size() == 0 || queueRate  < (threshold * consumersSet.size())) {
            if (logger.isDebugEnabled()) {
               logger.debug("Insufficient messages received on queue \"" + getName() + "\" to satisfy slow-consumer-threshold. Skipping inspection of consumer.");
            }

            return;
         }

         for (Consumer consumer : consumersSet) {
```